modules/core/shared/src/main/scala/lucuma/core/model/Target.scala

```diff
@@ -13,63 +13,230 @@ import lucuma.core.math._
+) extends Target
+
+object NonsiderealTarget extends NonsiderealTargetOptics {
+  implicit val NonsiderealTargetTargetEq: Eq[NonsiderealTarget] =
+    Eq.by(x => (x.name, x.ephemerisKey, x.magnitudes))
+
+  /**
+   * A nonsidereal target order based on ephemeris key.
+   *
+   * Not implicit.
+   */
+  val TrackOrder: Order[NonsiderealTarget] =
+    Order.by(t => (t.ephemerisKey, t.name, t.magnitudes.toList))
+
+  /**
+   * Nonsidereal targets ordered by name first and then ephemeris key.
+   *
+   * Not implicit.
+   */
+  val NameOrder: Order[NonsiderealTarget] =
+    Order.by(t => (t.name, t.ephemerisKey, t.magnitudes.toList))
+}
 
 object Target extends WithId('t') with TargetOptics {
-  implicit val TargetEq: Eq[Target] =
-    Eq.by(x => (x.name, x.track, x.magnitudes))
+
+  implicit val TargetEq: Eq[Target] = Eq.instance {
+    case (a @ SiderealTarget(_, _, _), b @ SiderealTarget(_, _, _))       => a === b
+    case (a @ NonsiderealTarget(_, _, _), b @ NonsiderealTarget(_, _, _)) => a === b
+    case _                                                                => false
+  }
 
   /**
-   * A target order based on tracking information.  For sidereal targets this
-   * roughly means by base coordinate without applying proper motion.  For
-   * non-sidereal this means by `EphemerisKey`.
+   * A target order based on tracking information. For sidereal targets this roughly means by base
+   * coordinate without applying proper motion. For non-sidereal this means by `EphemerisKey`.
    *
    * Not implicit.
    */
-  val TargetTrackOrder: Order[Target] =
-    Order.by(t => (t.track, t.name, t.magnitudes.toList))
+  val TrackOrder: Order[Target] =
+    Order.from {
+      case (a @ SiderealTarget(_, _, _), b @ SiderealTarget(_, _, _))       =>
+        SiderealTarget.TrackOrder.compare(a, b)
+      case (a @ NonsiderealTarget(_, _, _), b @ NonsiderealTarget(_, _, _)) =>
+        NonsiderealTarget.TrackOrder.compare(a, b)
+      case (NonsiderealTarget(_, _, _), _)                                  => -1
+      case _                                                                => 1
+    }
 
   /**
    * Targets ordered by name first and then tracking information.
    *
    * Not implicit.
    */
-  val TargetNameOrder: Order[Target] =
-    Order.by(t => (t.name, t.track, t.magnitudes.toList))
+  val NameOrder: Order[Target] =
+    Order.from {
+      case (a @ SiderealTarget(_, _, _), b @ SiderealTarget(_, _, _))       =>
+        SiderealTarget.NameOrder.compare(a, b)
+      case (a @ NonsiderealTarget(_, _, _), b @ NonsiderealTarget(_, _, _)) =>
+        NonsiderealTarget.NameOrder.compare(a, b)
+      case (NonsiderealTarget(_, _, _), _)                                  => -1
+      case _                                                                => 1
+    }
+}
+
+trait SiderealTargetOptics {
+
+  /** @group Optics */
+  lazy val name: Lens[SiderealTarget, NonEmptyString] =
+    Focus[SiderealTarget](_.name)
```

Comment:
Just out of curiosity, why do we make these `lazy`?  Are there initialization issues with these?